core/src/main/scala/kafka/log/LogCleanerManager.scala

```diff
@@ -365,6 +412,32 @@ private[log] class LogCleanerManager(val logDirs: Seq[File],
       }
     }
   }
+
+  def markPartitionUncleanable(logDir: String, partition: TopicPartition): Unit = {
+    inLock(lock) {
+      uncleanablePartitions.get(logDir) match {
+        case Some(partitions) =>
+          partitions.add(partition)
+        case None =>
+          uncleanablePartitions.put(logDir, mutable.Set(partition))
+      }
+    }
+  }
+
+  /*
+    Returns an immutable set of the uncleanable partitions for a given log directory
+   */
+  def uncleanablePartitions(logDir: String): Set[TopicPartition] = {
```

Comment:
This method doesn't seem to be used?

Reply:
It is used in the tests:
```
val uncleanablePartitions = cleaner.cleanerManager.uncleanablePartitions(uncleanableDirectory)
    assertTrue(uncleanablePartitions.contains(topicPartitions(0)))
    assertTrue(uncleanablePartitions.contains(topicPartitions(1)))
    assertFalse(uncleanablePartitions.contains(topicPartitions(2)))
```

We could either
a) keep the method, currently for test use only
b) remove the method and remove the assertion
c) remove the method and make the `uncleanablePartitions` field public so the test can access it

I've gone with b). Do confirm if you think this is the best approach

Reply:
We could do a) and make it visible at the package level and also add a comment that it's for testing only.